pkg/volume/util/nestedpendingoperations/nestedpendingoperations.go

```diff
@@ -244,23 +244,36 @@ func (grm *nestedPendingOperations) isOperationExists(key operationKey) (bool, i
 		return false, -1
 	}
 
+	opIndex := -1
 	for previousOpIndex, previousOp := range grm.operations {
 		volumeNameMatch := previousOp.key.volumeName == key.volumeName
 
 		podNameMatch := previousOp.key.podName == EmptyUniquePodName ||
 			key.podName == EmptyUniquePodName ||
 			previousOp.key.podName == key.podName
 
+		podNameExactMatch := previousOp.key.podName == key.podName
+
 		nodeNameMatch := previousOp.key.nodeName == EmptyNodeName ||
 			key.nodeName == EmptyNodeName ||
 			previousOp.key.nodeName == key.nodeName
 
+		nodeNameExactMatch := previousOp.key.nodeName == key.nodeName
+
 		if volumeNameMatch && podNameMatch && nodeNameMatch {
-			return true, previousOpIndex
+			// nonExactMatch pending first
+			if previousOp.operationPending {
+				return true, previousOpIndex
+			}
+			// nonExactMatch with no pending, set opIndex to the first nonExactMatch
+			// exactMatch can override opIndex to expected
+			if opIndex == -1 || (podNameExactMatch && nodeNameExactMatch) {
+				opIndex = previousOpIndex
+			}
```

Comment:
There is another correctness issue I see here, if:

a. {pod1,vol1, umount} -> enqueued
b. {pod2, vol1, umount} -> enqueued
c. {pod1, vol1, umount, pending=false (fail} {pod2, vol1, umount, pending=true}
d. {pod3, vol1, mount} -> enqued

In step#d - mount operation replaces unmount operation in step#c at index-0. Whereas strictly speaking, mount function contract says, we should not call mount operation if any other operation on same volume is pending (even if it is happening on a different pod).

So correct me if I am wrong but this PR does not fixes that behavior. We could still call mount on same volume, while another pod is trying to unmount the volume and unmount operation is pending.

Reply:
err - actually I think I was wrong. We do prefer pending operations first and hence we don't break that contract.